zerver/views/registration.py

```diff
@@ -243,6 +246,21 @@ def get_selected_realm_default_language_name(
     return get_language_name(prereg_realm.default_language)
 
 
+def generate_demo_realm_subdomain() -> str:
+    available_subdomain = False
+    demo_subdomain = ""
+    while not available_subdomain:
+        letters = "".join(random.SystemRandom().choice(string.ascii_lowercase) for _ in range(4))
+        digits = "".join(random.SystemRandom().choice(string.digits) for _ in range(4))
+        demo_subdomain = f"demo-{letters}{digits}"
+        try:
+            check_subdomain_available(demo_subdomain)
+            available_subdomain = True
+        except ValidationError:  # nocoverage
+            continue
+    return demo_subdomain
```

Comment:
This is basically the same code that we've been using for demo organization subdomains in the dev environment, except in a while loop so that we definitely get an available subdomain.

Not sure if we want to either make this more or less random. I could imagine something a dashed string of random English words that starts with demo being perhaps nicer for users, e.g., random subdomains like "demo-apple-ladder-purple".

Reply:
Yeah, we can use a package like https://github.com/the-lay/py-friendly-words for this. @alya FYI